src/madengine/tools/update_perf_csv.py

```diff
@@ -116,21 +114,24 @@ def handle_multiple_results(
     # add results to perf.csv
     for r in multiple_results_df.to_dict(orient="records"):
         row = common_info_json.copy()
-        row["model"] = model_name + "_" + str(r["model"])
-        row["performance"] = r["performance"]
-        row["metric"] = r["metric"]
+        model = r.pop("model")
+        row["model"] = model_name + "_" + str(model)
+        row.update(r)
```

Comment:
This allows us to add additional columns to the perf csv; this is really the major change from this PR, the other changes just support this e.g. reordering the columns in the final_multiple_results_df to put any additional columns at the end.